tests/functional/defer_state/test_defer_state.py

```diff
@@ -345,3 +350,63 @@ def test_defer_state_flag(self, project, unique_schema, other_schema):
         assert results.results[0].status == RunStatus.Success
         assert results.results[0].node.name == "table_model"
         assert results.results[0].adapter_response["rows_affected"] == 2
+
+
+get_schema_name_sql = """
+{% macro generate_schema_name(custom_schema_name, node) -%}
+    {%- set default_schema = target.schema -%}
+    {%- if custom_schema_name is not none -%}
+        {{ return(default_schema ~ '_' ~ custom_schema_name|trim) }}
+    -- put seeds into a separate schema in "prod", to verify that cloning in "dev" still works
+    {%- elif target.name == 'default' and node.resource_type == 'seed' -%}
+        {{ return(default_schema ~ '_' ~ 'seeds') }}
+    {%- else -%}
+        {{ return(default_schema) }}
+    {%- endif -%}
+{%- endmacro %}
+"""
+
+
```

Comment:
To circle back to talk of test placement does creating two base classes ex. `BaseClonePossible` and `BaseCloneNotPossible` and define like the tests below to look for the correct types `view or table` so users can pull in one or both options to test? or potentially move both those to their own file to be the adapter zone tests? as I don't believe `defer_state` is pulled into all adapters

Reply:
will make some mock stuff for snowflake pr we can look at